Packages/ChatSecure/ChatSecure/Classes/WebRTCJanus/Janus.swift

```diff
@@ -61,85 +61,92 @@ class Janus: NSObject {
+						  "plugin": plugin.pluginName,
+						  "opaque_id": plugin.opaqueId,
+						  "transaction": transaction,
+						  "session_id": NSNumber(value: self.sessionId)] as [String: Any]
+			if let token = self.token {
+				params["token"] = token
+			}
+			if let secret = self.apiSecret {
+				params["apisecret"] = secret
+			}
+			let reqCallback: JanusRequestCallback = { [weak self] msg in
+				guard let self = self else { return }
+				self.handleAttachPluginCallback(plugin: plugin,
+												pluginCallback: callback,
+												msg: msg)
+			}
+			self.janusTransactions.updateValue(reqCallback, forKey: transaction)
+			if self.sessionId == 0 {
+				self.delayReq.append(params)
+				if self.janusWebSocket == nil {
+					self.janusWebSocket = JanusWebSocket(withServer: self.server)
+					self.janusWebSocket?.delegate = self
+					self.janusWebSocket?.start()
+				}
+			} else {
+				self.janusWebSocket?.send(message: params)
 			}
-		} else {
-			janusWebSocket?.send(message: params)
 		}
 	}
 	
 	// MARK: - Detach WebSocket
 	func detachPlugin(plugin: JanusPluginHandleProtocol, callback: DetachCallback?) {
-		pluginHandlers.removeValue(forKey: plugin.handleId)
-		
-		if sessionId == 0 || !plugin.attached || janusWebSocket != nil {
-			if let reqCallback = callback {
-				reqCallback(nil)
-			}
-			return
-		}
-		
-		var transaction = randomString(withLength: 12)
-		while janusTransactions.keys.contains(transaction) {
-			transaction = randomString(withLength: 12)
-		}
-		
-		var params = ["janus": JanusMessage.detach,
-					  "handle_id": plugin.handleId,
-					  "transaction": transaction,
-					  "session_id": NSNumber(value: sessionId)] as [String: Any]
-		if let token = token {
-			params["token"] = token
-		}
-		if let secret = apiSecret {
-			params["apisecret"] = secret
-		}
-		let callbackJanus: JanusRequestCallback = { msg in
-			if let reqCallback = callback {
-				if let janus = msg["janus"] as? String, janus == "success" {
+		mainque.async {
+			self.pluginHandlers.removeValue(forKey: plugin.handleId)
+			
+			if self.sessionId == 0 || !plugin.attached || self.janusWebSocket != nil {
+				if let reqCallback = callback {
 					reqCallback(nil)
-				} else if let janus = msg["janus"] as? String, janus == "error" {
-					reqCallback(JanusResultError.funcErr(msg: "Detach error"))
+				}
+				return
+			}
+			
+			var transaction = randomString(withLength: 12)
+			while self.janusTransactions.keys.contains(transaction) {
+				transaction = randomString(withLength: 12)
+			}
+			
+			var params = ["janus": JanusMessage.detach,
+						  "handle_id": plugin.handleId,
+						  "transaction": transaction,
+						  "session_id": NSNumber(value: self.sessionId)] as [String: Any]
+			if let token = self.token {
+				params["token"] = token
+			}
+			if let secret = self.apiSecret {
+				params["apisecret"] = secret
+			}
+			let callbackJanus: JanusRequestCallback = {[weak self] msg in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/Packages/ChatSecure/ChatSecure/Classes/WebRTCJanus/Janus.swift#L137"></span>Unused reference self in a capture list should be removed.<br />
<code>unused_capture_list</code> <code>Janus.swift:137</code></td>
    </tr>
  </tbody>
</table>